ndnsd/discovery/service-discovery.cpp

```diff
@@ -193,31 +220,27 @@ ServiceDiscovery::makeDataContent()
 void
 ServiceDiscovery::sendData(const ndn::Name& name)
 {
-  NDNSD_LOG_INFO("Sending data for: " << name);
+  NDN_LOG_INFO("Prepare to send data for name: " << name);
+
   auto timeDiff = ndn::time::system_clock::now() - m_producerState.publishTimestamp;
-  auto status = (timeDiff >= m_producerState.serviceLifetime*1000)
-                          ? EXPIRED : ACTIVE;
+  auto timeToExpire = ndn::time::duration_cast<ndn::time::seconds>(timeDiff);
 
-  std::shared_ptr<ndn::Data> replyData = std::make_shared<ndn::Data>(name);
-  replyData->setFreshnessPeriod(1_ms);
+  int status = (timeToExpire > m_producerState.serviceLifetime) ? EXPIRED : ACTIVE;
 
-  auto dataContent = makeDataContent();
-  auto& data = wireEncode(dataContent, status);
-  replyData->setContent(data);
-  m_keyChain.sign(*replyData);
+  ndn::Data replyData(name);
+  replyData.setFreshnessPeriod(1_ms);
 
-  try
-  {
-    m_face.put(*replyData);
-  }
-  catch (const std::exception& ex) {
-    std::cerr << ex.what() << std::endl;
-  }
+  auto dataContent = makeDataContent();
```

Comment:
Perhaps move the content of makeDataContent to wireEncode?

Reply:
So that you can remove arguments from wireEncode and get the status and make the content within it. Simpliyfing this function.
Also if status and data has not changed then the old m_wire should suffice.
So probably status can be a member of the class.

Reply:
yes, this can be done.